### ISSUE TITLE
Dialog: focusTrap not working

### ISSUE THREAD

**Issue by basvandenberg** [primefaces/primeng#12957]
### Describe the bug

The dialog component has a focusTrap property that is set to true by default, "When enabled, can only focus on elements inside the dialog."

In PrimeNg v15 the focus trap doesn't work. If you open a dialog and press tab, the focus stays on the elements behind the dialog. This can also be observed on https://primeng.org/dialog (v15.4.1).

### Environment

The behavior can be observed on the primeng website for version 15.4.1:
https://primeng.org/dialog

### Angular version

15.0.2

### PrimeNG version

15.4.1

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

18.16.0

### Browser(s)

Chrome 112

### Steps to reproduce the behavior

1. Open dialog on page https://primeng.org/dialog#basic
2. Press tab button several times

### Expected behavior

When a dialog is open and the focusTrap property is set to true, when pressing tab, focus should stay on elements within the dialog.

**Comment by GitJacek** [comment]
Any update/workaround on this one?

**Comment by yuyu668668** [comment]
I have the same problem.

**Comment by pepaitdev** [comment]
me too
